File: array/1304_find_n_unique_integers_sum_up_to_zero/1304_find_n_unique_integers_sum_up_to_zero.go
package main

import "fmt"

func sumZero(n int) []int {
	res := make([]int, n)
	for num, i := n/2, 0; i < n/2; i, num = i+1, num-1 {
		res[i] = -num
		res[n/2+i] = num
	}

	return res
}

func main() {
	n := 5
	res := sumZero(n)
	fmt.Println(res)
}

File: array/1304_find_n_unique_integers_sum_up_to_zero/1304_find_n_unique_integers_sum_up_to_zero.py
from typing import List


class Solution:
    def sumZero(self, n: int) -> List[int]:
        res = []
        if n%2 == 1:
            res.append(0)
        for i in range(1, n//2+1):
            res.append(i)
            res.append(-i)
        return res


n = 5
res = Solution().sumZero(n)
print(res)